### ISSUE TITLE
with_default only works for subset of linters

### ISSUE THREAD

**Issue by jackwasey** [jimhester/lintr#56]
```
works <- with_defaults(object_camel_case_linter=NULL)
fails <- with_defaults(camel_case_linter=NULL) # ignores the setting
```

**Comment by jimhester** [comment]
What `lintr` version are you using for this?

Best way to get the exact SHA is

```r
devtools::session_info()
```

I shorted the names of the object linters in 5532edccfefdb6a3f0f15053fa0b172acbd8eefa, so if you are using a version prior to that only the former will work, if after only the latter.

**Comment by jackwasey** [comment]
My bad, I was behind at 0378b0b. Updating also fixed #57... Thanks!
